tool/builder/lib/sdks_finder.dart

```diff
@@ -0,0 +1,345 @@
+        MediaPipeSdk.libtext => 'mediapipe-task-text',
+        MediaPipeSdk.libvision => 'mediapipe-task-vision',
+      };
+}
+
+/// Scans the GCS buckets where MediaPipe SDKs are stored, identifies the latest
+/// builds for each supported build target, and writes the output to a
+/// designated file.
+///
+/// [SdksFinderCommand] depends on `gsutil` and a Google-corp account that has
+/// permissions to read the necessary buckets, so the idea is to run this
+/// command on Googlers' work machines whenever MediaPipe rebuilds SDKs. The
+/// manual nature of this command is suboptimal, but there is no precedent for
+/// fully automating this portion of the MediaPipe release process, so this
+/// command helps by automating a portion of the task.
+///
+/// The cache-busting mechanism of Flutter's native assets feature is a hash
+/// of the contents of any build dependencies, so if this command leads to any
+/// new build state, this must be reflected by *some change* to the associated
+/// library's `build.dart` script.
+///
+/// Operationally, [SdksFinderCommand]'s implementation involves orchestrating
+/// one [_OsFinder] instance for each supported [OS] value, which in turn
+/// identifies the correct build that has been uploaded to GCS by MediaPipe
+/// release machinery.
+///
+/// Usage:
+/// ```sh
+///   $ cd path/to/flutter-mediapipe
+///   $ dart tool/builder/bin/main.dart sdks [-v]
+/// ```
+class SdksFinderCommand extends Command with RepoFinderMixin {
+  SdksFinderCommand() {
+    addVerboseOption(argParser);
+  }
+  @override
+  String description =
+      'Downloads the appropriate MediaPipe SDKs for the current build target';
+
+  @override
+  String name = 'sdks';
+
+  static const _gcsPrefix = 'https://storage.googleapis.com';
+
+  /// Google Storage bucket which houses all MediaPipe SDK uploads.
+  static const _bucketName = 'mediapipe-nightly-public/prod/mediapipe';
+
+  final _finders = <_OsFinder>[
+    _OsFinder(OS.android),
+    _OsFinder(OS.macOS),
+    // TODO: Add other values as their support is ready
+  ];
+
+  @override
+  Future<void> run() async {
+    setUpLogging();
+    _checkGsUtil();
+    final results = _SdkLocations();
+
+    for (final finder in _finders) {
+      await for (final sdkLocation in finder.find()) {
+        results.add(sdkLocation);
+      }
+    }
+    for (final MediaPipeSdk sdk in MediaPipeSdk.values) {
+      _log.info('Saving locations for ${sdk.name}');
+      _writeResults(sdk, results.toMap(sdk));
+    }
+  }
+
+  void _writeResults(MediaPipeSdk sdk, _FlatResults results) {
+    final file = _getOutputFile(sdk);
+    _log.fine('Writing data to "${file.absolute.path}"');
+    var encoder = JsonEncoder.withIndent('  ');
+    file.writeAsStringSync('''// Generated file. Do not manually edit.
+final Map<String, Map<String, String>> sdkDownloadUrls = ${encoder.convert(results).replaceAll('"', "'")};
+''');
+  }
+
+  File _getOutputFile(MediaPipeSdk sdk) {
+    return File(
```

Comment:
Does this handle if the file already exists? Or does not exist yet?

Reply:
Dart's IO library is well written on this front. `file.writeAsString()` elegantly handles both situations 🎉